meson.build

```
@@ -0,0 +1,519 @@
+  sources += ['lib/aix-device.c']
+elif host_machine.system() == 'netbsd'
+  conf.set('PCI_HAVE_PM_NBSD_LIBPCI', 1)
+  conf.set('PCI_HAVE_PM_MMIO_CONF', 1)
+  conf.set('PCI_HAVE_PM_ECAM', 1)
+  conf.set('PCI_PATH_NBSD_DEVICE', '/dev/pci0')
+  conf.set('PCI_PATH_DEVMEM_DEVICE', '/dev/mem')
+  conf.set('PCI_PATH_ACPI_MCFG', '')
+  conf.set('PCI_PATH_EFI_SYSTAB', '')
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  sources += [
+    'lib/mmio-ports.c',
+    'lib/ecam.c',
+    'lib/nbsd-libpci.c',
+  ]
+elif host_machine.system() == 'gnu'
+  conf.set('PCI_HAVE_PM_HURD_CONF', 1)
+  conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+  sources += [
+    'lib/hurd.c',
+  ]
+  + i386_ports_srcs
+elif host_machine.system() == 'djgpp'
+  conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+  sources += i386_port_srcs
+elif host_machine.system() == 'cygwin'
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  conf.set('PCI_HAVE_PM_WIN32_CFGMGR32', 1)
+  conf.set('PCI_HAVE_PM_WIN32_KLDBG', 1)
+  conf.set('PCI_HAVE_PM_WIN32_SYSDBG', 1)
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += [
+      'lib/emulated.c',
+      'lib/win32-cfgmgr32.c',
+      'lib/win32-kldbg.c',
+      'lib/win32-sysdbg.c',
+    ]
+    + i386_port_srcs
+  endif
+  dependencies += [ cc.find_library('ioperm', required: true) ]
+elif host_machine.system() == 'windows'
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  conf.set('PCI_HAVE_PM_WIN32_CFGMGR32', 1)
+  conf.set('PCI_HAVE_PM_WIN32_KLDBG', 1)
+  conf.set('PCI_HAVE_PM_WIN32_SYSDBG', 1)
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += [
+      'lib/emulated.c',
+      'lib/win32-cfgmgr32.c',
+      'lib/win32-kldbg.c',
+      'lib/win32-sysdbg.c',
+    ]
+    + i386_ports_srcs
+  endif
+  dependencies += [ cc.find_library('advapi32', required: true) ]
+elif host_machine.system() == 'haiku'
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += i386_ports_srcs
+  endif
+  conf.set('PCI_HAVE_STDINT_H', 1)
+elif host_machine.system() == 'sylixos'
+  conf.set_quoted('PCI_PATH_SYLIXOS_DEVICE', '/proc/pci')
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  conf.set('PCI_HAVE_PM_SYLIXOS_DEVICE', 1)
+  IDSDIR = get_option('prefix') / get_option('sysconfdir') / 'pci'
+  sources += ['lib/sylixos-device.c']
+else
+  error('Unfortunately, your OS is not supported by the PCI library')
+endif
+
+conf.set_quoted('PCI_PATH_IDS_DIR', IDSDIR)
+
+conf.set10('PCI_COMPRESSED_IDS', zlib_dep.found())
+if zlib_dep.found()
+  conf.set_quoted('PCI_IDS', 'pci.ids.gz')
+  pci_ids = 'pci.ids.gz'
+else
+  conf.set_quoted('PCI_IDS', 'pci.ids')
+  pci_ids = 'pci.ids'
+endif
+
+if (get_option('with-dns').enabled() or
+  get_option('with-dns').auto() and host_machine.system() != 'windows')
+
+  assert(cc.has_function('res_query', dependencies:dns_dep))
```

```suggestion
  assert(cc.has_function('res_query', dependencies:dns_dep), 'missing required function: res_query')
```

will drop the minimum required version from 0.53.0 to 0.50.0